### PR TITLE
Show confirmation modal before deleting order messages

### LINKED ISSUE #18415: Missing modal to confirm or cancel the deletion of an order message (in bulk)
#### Describe the bug

It was planned to add a modal to confirm or cancel the bulk deletion during the migration of the page but it was not done #10555

#### Expected behavior

Add a modal to confirm or cancel the deletion of many rows

#### Additional information

* PrestaShop version: 1.7.7.x
* PHP version: N/A

### PR DESCRIPTION
**Pull request by zuk3975**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | 1.7.7.x
| Description?  | Add confirmation modal to be shown before deleting order messages in Customer Service > Order messages. Add in bulk action only (It was said that single row action is supposed to go to 1.7.8 and its already started in other pr). Technically it implements BulkDeleteActionTrait grid definition and uses SubmitBulkActionExtension in js.
| Type?         | bug fix 
| Category?     | BO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | #18415 
| How to test?  | Go to Customer services > Order messages and try to delete order message in bulk action. You should see following modal to confirm your action. If you press Delete, your action should be processed, else it should not delete your selection. Screenshot from 2020-04-16 09-41-13

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/18641)

### CODE REVIEW

File: admin-dev/themes/new-theme/js/pages/order_message/index.js [hunk 1]
```diff
@@ -29,11 +29,11 @@ import ExportToSqlManagerExtension from '@components/grid/extension/export-to-sq
 import FiltersResetExtension from '@components/grid/extension/filters-reset-extension';
 import SortingExtension from '@components/grid/extension/sorting-extension';
 import LinkRowActionExtension from '@components/grid/extension/link-row-action-extension';
-import SubmitBulkExtension from '@components/grid/extension/submit-bulk-action-extension';
 import BulkActionCheckboxExtension from '@components/grid/extension/bulk-action-checkbox-extension';
 import SubmitRowActionExtension from '@components/grid/extension/action/row/submit-row-action-extension';
 import FiltersSubmitButtonEnablerExtension
   from '@components/grid/extension/filters-submit-button-enabler-extension';
+import SubmitBulkActionExtension from '../../components/grid/extension/submit-bulk-action-extension';
```
Comment (Progi1984) [inline admin-dev/themes/new-theme/js/pages/order_message/index.js]:
```suggestion
import SubmitBulkActionExtension from '@components/grid/extension/submit-bulk-action-extension';
```

File: src/Core/Grid/Definition/Factory/OrderMessageGridDefinitionFactory.php [hunk 2]
```diff
@@ -185,6 +192,12 @@ protected function getBulkActions()
                     ->setOptions([
                         'submit_route' => 'admin_order_messages_bulk_delete',
                         'confirm_message' => $this->trans('Delete selected items?', [], 'Admin.Notifications.Warning'),
+                        'modal_options' => new ModalOptions([
```
Comment (jolelievre) [inline src/Core/Grid/Definition/Factory/OrderMessageGridDefinitionFactory.php]: You should use the `BulkDeleteActionTrait::buildBulkDeleteAction` here, it's easier and more consistent with other definitions

**Review by jolelievre — CHANGES_REQUESTED** [review]
Hi @zuk3975 
Sorry but the modal on row action is not in the scope of 177, it's considered a new feature and was not ready before feature freeze. So only modal on bulk action are integrated in 177

The integration for 178 of the row action has stared though, it's based on a similar trait as `BulkDeleteActionTrait` to easily add the action and only update the grid definition

**Comment by zuk3975** [comment]
> Hi @zuk3975
> Sorry but the modal on row action is not in the scope of 177, it's considered a new feature and was not ready before feature freeze. So only modal on bulk action are integrated in 177
> 
> The integration for 178 of the row action has stared though, it's based on a similar trait as `BulkDeleteActionTrait` to easily add the action and only update the grid definition

ok, I've reverted my changes for single row. 
Also resolved all other feedbacks.

**Review by Progi1984 — APPROVED** [review]

**Review by jolelievre — APPROVED** [review]

**Comment by SD1982** [comment]
This PR correctly add the order message bulk delete confirmation modal. Thanks @zuk3975

**Comment by Progi1984** [comment]
Thanks @zuk3975

**Pull request merged**